src/driver/amdxdna/aie2_event_trace.c

```diff
@@ -0,0 +1,326 @@
+#include <drm/drm_cache.h>
+#include "aie2_msg_priv.h"
+#include "aie2_pci.h"
+#include "amdxdna_trace.h"
+#include "amdxdna_mailbox.h"
+
+struct event_trace_req_buf {
+	struct amdxdna_dev_hdl   *ndev;
+	struct workqueue_struct  *wq;
+	struct work_struct       work;
+	u8                       *fw_log_buf;
+	u8                       *buf;
+	u64                      dram_buffer_address;
+	u64                      resp_timestamp;
+	u64                      sys_start_time;
+	u32                      dram_buffer_size;
+	int                      log_ch_irq;
+	bool                     event_trace_enabled;
+};
+
+struct trace_event_metadata {
+	u64 tail_offset;
+	u64 head_offset;
+	u32 padding[12];
+};
+
+struct trace_event_log_data {
+	u64 counter;
+	u16 payload_hi;
+	u16 type;
+	u32 payload_low;
+};
+
+static void clear_event_trace_msix(struct amdxdna_dev_hdl *ndev)
+{
+	/* Clear the log buffer interrupt */
+	writel(0, (void *)((u64)ndev->mbox_base + (u64)LOG_BUF_MB_IOHUB_PTR));
+}
+
+static int aie2_is_event_trace_supported_on_dev(struct amdxdna_dev_hdl *ndev)
+{
+	struct pci_dev *pdev = to_pci_dev(ndev->xdna->ddev.dev);
+
+	XDNA_DBG(ndev->xdna, "Dev id: 0x%x, Dev rev: 0x%x\n", pdev->device, pdev->revision);
+	return (pdev->device == 0x17f0 && pdev->revision >= 0x10);
+}
+
+static u32 aie2_get_trace_event_content(struct event_trace_req_buf *trace_req_buf)
+{
+	struct amdxdna_dev_hdl *ndev = trace_req_buf->ndev;
+	struct trace_event_metadata *trace_metadata;
+	u8 *sysBuf = trace_req_buf->buf;
+	u8 *kern_buf = trace_req_buf->fw_log_buf;
+	u32 rd_ptr, wr_ptr, wr_ptr_wrap;
+	u32 total_log_size = 0, log_size = 0;
+	u32 rb_size, offset = 0;
+
+	rb_size = TRACE_EVENT_BUFFER_SIZE - TRACE_EVENT_BUFFER_METADATA_SIZE;
+	if (rb_size == 0)
+		return 0;
+
+	trace_metadata = (struct trace_event_metadata *)(sysBuf + rb_size);
+	rd_ptr = (u32)(trace_metadata->head_offset % rb_size);
+	wr_ptr = (u32)(trace_metadata->tail_offset);
+	wr_ptr_wrap = wr_ptr % rb_size;
+
+	/* Update the Ring Buffer read pointer */
+	trace_metadata->head_offset = wr_ptr;
+
+	do {
+		if (wr_ptr_wrap > rd_ptr)
+			log_size = wr_ptr_wrap - rd_ptr;
+		else if (wr_ptr_wrap < rd_ptr)
+			log_size = rb_size - rd_ptr;
+		else
+			return 0;
+
+		if (log_size > rb_size) {
+			XDNA_ERR(ndev->xdna, "log_size > rb_size");
+			return 0;
+		}
+		/* Copy the ring buffer content to kernel buffer */
+		memcpy(kern_buf + offset, (u8 *)(sysBuf + rd_ptr), log_size);
+
+		offset += log_size;
+		total_log_size += log_size;
+		rd_ptr = (rd_ptr + log_size) % rb_size;
+	} while (rd_ptr < wr_ptr_wrap);
+
+	return total_log_size;
+}
+
+static void aie2_print_trace_event_log(struct amdxdna_dev_hdl *ndev)
+{
+	struct event_trace_req_buf *trace_req_buf;
+	struct trace_event_log_data *log_content;
+	u64 payload;
+	u32 log_size;
+
+	trace_req_buf = ndev->event_trace_req;
+	log_size = aie2_get_trace_event_content(trace_req_buf);
+	XDNA_DBG(ndev->xdna, "FW log size in bytes %u", log_size);
+
+	if (log_size) {
```

Comment:
Use below coding style to avoid indentation.
```suggestion
	if (!log_size)
	    return;
	    
	// Read and print log
```